Adds fidget spinners

deathride58 opened the pull request:
:cl: deathride58
add: Adds fidget spinners. They can be found in arcade machines and cargo's toy crates!
add: Adds syndi-spinners. They're red, which means they're actually somewhat robust. Costs a super expensive 1 TC! They're also capable of acting as traps if spun and dropped.
/:cl:

I spent two hours wanting to die so no one else has to.

Gif:
![ss13fidgetorigupd8](https://cloud.githubusercontent.com/assets/6356337/26181549/fd4fec04-3b3e-11e7-911a-0d6f45610ca3.gif)

Left: Fidget Spinner, Right: Syndi-Spinner
Top: Resting, Bottom: Spinning

deathride58 committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

deathride58 committed:
Adds fidget spinners

WJohn commented:
Inevitable that this meme would reach us.

obscolene commented:
this PR is like picasso painting with human feces

PKPenguin321 commented:
8 TCs for a just worse throwing star

deathride58 committed:
Fixes travis

juanototo2 commented:
either make it 8 for a box of 12 or make it cost literally one tc if PKP is roasting you like that

deathride58 committed:
Nerfs uplink price

deathride58 commented:
Nerfed uplink price to 1 TC

deathride58 committed:
Adds name to syndi-spinner

optimumtact commented:
@deathride58 screenshots, I have some from a spriter, along with an animation for them

I can give them to you if they look better.

deathride58 commented:
@optimumtact Added screenshot of the items to the OP. I don't any have in-hand sprites for 'em, unfortunately.

ShadowDeath6 commented on code/modules/uplink/uplink_item.dm:
```diff
@@ -1319,6 +1319,13 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
+/datum/uplink_item/badass/syndiespinner
+	name = "Syndicate Fidget Spinner"
+	desc = "This fidget spinner is designed to be a little more robust than others like it. \
+			Its bearings are much smoother to allow for faster spinning, and its blades are actual blades."
```
"Its blades are actually blades"
Could be reworded, "its blades are actually sharp" would be better imo

deathride58 replied:
👌

deathride58 committed:
Words uplink item description better.

optimumtact commented:
I'll upload my DMI's tonight to compare

dannno commented:
not very funny

ExcessiveUseOfCobblestone commented:
I don't mind the fluff item because xd maymays but why does a syndicate variant need to exist?

Also these should be contraband simply because they make people go insane.

Tacolizard commented:
the sprites would look better if they were top-down instead of being perspective-shifted. The still fidget spinner is hard to recognize.

InsaneHyena commented:
It's a fad that will pass in a month or so, does it really need to be included into the code?

optimumtact commented:
Yes becuase I like it

optimumtact commented:
https://file.house/Trme.dmi here are the sprites I have, top down and different spin animation

Time-Green commented:
Make it do braindamage upon use

banthebantz commented:
make them either heal or cause brain damage

Dorsisdwarf commented:
why do we need to add every passing craze to ss13

calzilla1 commented:
Yes apparently

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	force = 1
+	throwforce = 3
+	var/force_spinning = 3
+	var/throwforce_spinning = 5
```
new vars after parent vars

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	throw_speed = 3
+	throw_range = 5
+	attack_verb = list("slapped", "tapped", "papped")
+	var/spinning = 0
```
FALSE

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	throw_range = 5
+	attack_verb = list("slapped", "tapped", "papped")
+	var/spinning = 0
+	var/syndispinner = 0
```
This var is not necessary

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	
+/obj/item/weapon/fidgetspinner/proc/stop_spin(mob/user)
+	if(spinning)
+		spinning = !spinning
```
= FALSE

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	else
+		return
+		
+/obj/item/weapon/fidgetspinner/attack(mob/M, mob/user, def_zone)
```
attack_hand

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+		return
+		
+/obj/item/weapon/fidgetspinner/attack(mob/M, mob/user, def_zone)
+	..()
```
. = ..()

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+		stop_spin(user)
+
+/obj/item/weapon/fidgetspinner/throw_impact(atom/target)
+	..()
```
. = ..()

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+		stop_spin(thrownby)
+		
+/obj/item/weapon/fidgetspinner/attackby(obj/item/I, mob/user, params)
+	..()
```
This will damage the spinner all the time i hope you realize

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	if(spinning)
+		icon_state = "[initial(icon_state)]_spin"
+	else
+		icon_state = "[initial(icon_state)]"
```
You can oneline this proc with a ternery

tortellinitony replied:
Just make it [initial(icon_state)]_[spinning]

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+		force = force_spinning
+		throwforce = throwforce_spinning
+		if(syndispinner)
+			embed_chance = 50
```
put this in the child's proc

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+	if(spinning)
+		user.visible_message("[user] starts spinning [user.p_their()] [src]...", "<span class='notice'>You start spinning [src]...</span>")
+		handle_spin(user)
+	else
```
Create a new proc called hand stop spin and specialize it for the generic spinner and syndicate spinner that will handle the stuff in this else

Cyberboss commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,86 @@
+/obj/item/weapon/fidgetspinner/syndicatespinner
+	name = "syndi-spinner"
+	desc = "It's a Donk Co. branded Syndi-Spinner. Instead of having thick, dull blades like fidget spinners typically do, this one has thin, sharp blades."
+	syndispinner = 1
```
remove

tortellinitony commented:
I wouldn't care about these as a feature, but the sprites are really shitty, no offense.

tortellinitony commented:
Also, since you made the normal spinner white, it should be colorable.

AlexCrimson commented:
If we add ever single damn meme/fad that pops up, eventually this server will be complete crap. Nothing but memers acting like idiots all the time. Its bad enough already with some idiot posting copypasta memes in OOC every other round.

Bawhoppen commented:
@AlexCrimson we shouldn't limit the code based on how players act, we instead limit the players with rules (but nobody seems to want to try and do that very much)

Cheridan commented:
>If we add ever single damn meme/fad that pops up, eventually this server will be complete crap. 

oops

Grazyn commented:
And we STILL don't have hoverboards

AnturK commented:
Ambivalent on the toy but let's not dilute uplink with 20 minimal variations of the same thing (aka throwing stars) please.

Shapsy commented:
Uh, does anyone else think the sprite looks like a dick?

Shapsy commented:
What an interestingly shaped sprite

deathride58 committed:
Syndi-spinners can now act similar to bear traps, spinners can now cause brain damage, cleaned code, and more

deathride58 commented:
Updated sprites to use @optimumtact 's sprites.

When fidget spinners stop spinning, whoever's responsible will receive brain damage. 3 brain damage if they have less than 30 brain damage, 1 brain damage if they're over that. When spinning a spinner, people with disabilities now heal 1 brain damage.

Fidget spinners are now unusable by non-human mobs and those incapable of fine manipulation, as such a foolish contraption is below their interests.

Syndi-spinners now have an extra function, allowing them to act in a similar way to bear traps. When they're spinning, anyone that tries to walk over them will trip and take damage.

Also cleaned up code quite a bit

banthebantz commented:
is it just me or is there a swastika in that spinning gif?

Tacolizard commented:
yuck those new sprites are 10 thousand times worse. Just make non perspective-shifted versions of the original sprite.

Tacolizard commented:
kevinz how the hell did you +1 this pr

ShadowDeath6 commented:
The spinning sprites look really bad now I prefer the old ones

deathride58 commented:
Even shittier sprites comin right up

deathride58 committed:
Third time's the charm (3rd fidget spinner sprites)

Grazyn commented:
They look like MiniDisc now

ChangelingRain commented:
you know, I thought the first sprites were kind of weird, but you and oranges have shown me just how much worse it could be.
go back to those, please.

ChangelingRain commented on code/game/objects/items/fidgetspinners.dm:
```diff
@@ -0,0 +1,123 @@
+				return
+			else
+				L.apply_damage(throwforce,BRUTE,hitzone)
+				L.Stun(15)
```
holy fuck no make this like 4 or 5, this is 30 seconds of stun

deathride58 replied:
I'm buffing it to 1800

deathride58 committed:
Rolls spinner sprite back to original version and nerfs the syndi-spinner's stun

deathride58 committed:
Nerfs runtimes when spinning spinners meet explosions

Repukan commented:
Looks like a penis

WJohn commented:
Sprites are nice.

Also wowie this PR blew up.

duncathan reviewed (dismissed):
i hate this

Cyberboss commented:
Review abuse

On May 18, 2017 12:13 PM, "duncathan salt" <notifications@github.com> wrote:

> *@duncathan* requested changes on this pull request.
>
> i hate this
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/27349#pullrequestreview-38986376>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywelgoM_kMlmL3mfp-Sd-VfUY-gjxAks5r7G4agaJpZM4NdSSC>
> .
>

duncathan commented:
imo all maintainers should use reviews like this it would make decisions easier

Iamgoofball commented:
Review abuse

BeeSting12 commented:
*dab next guys!

deathride58 committed:
Fixes conflict w/ toy his grace pr

deathride58 committed:
Merge branch 'master' into fidgetspinners

AnturK commented:
I'll be the no fun allowed maintainer today.

Cyberboss commented:
wtf I love @optimumtact now
